examples/README.md

```diff
@@ -0,0 +1,40 @@
+## Structure
+
+For the simplest examples without using any helping code (see `framework.rs` here), check out:
+  - `hello ` for printing adapter information
+  - `hello-triangle` for graphics and presentation
+  - `hello-compute` for pure computing
+
+Notably, `capture` example show rendering without a surface/window. It reads back the contents and saves them to a file.
+
+All framework-based examples render to the window.
+
+## Feature matrix
+| Feature                | boids  | cube   | mipmap | msaa-line | shadow | skybox | texture-arrays | water  |
+| ---------------------- | ------ | ------ | ------ | --------- | ------ | ------ | -------------- | ------ |
+| vertex attribute       | :star: | :star: | :star: | :star:    | :star: |        | :star:         | :star: |
+| instancing             | :star: |        |        |           |        |        |                |        |
+| lines and points       |        |        |        | :star:    |        |        |                |        |
+| sampled color texture  | :star: | :star: | :star: |           |        | :star: | :star:         | :star: |
+| storage texture        | :star: |        |        |           |        |        |                |        |
+| binding array          |        |        |        |           |        |        | :star:         |        |
+| comparison sampler     |        |        |        |           | :star: |        |                |        |
+| subresource view       |        |        | :star: |           | :star: |        |                |        |
+| cubemap                |        |        |        |           |        | :star: |                |        |
+| multisampling          |        |        |        | :star:    |        |        |                |        |
+| off-screen rendering   |        |        |        |           | :star: |        |                | :star: |
+| stencil test           |        |        |        |           |        |        |                |        |
+| depth test             |        |        |        |           | :star: |        |                | :star: |
+| read-only depth        |        |        |        |           |        |        |                | :star: |
+| blending               |        |        |        |           |        |        |                | :star: |
+| render bundle          |        |        |        | :star:    |        |        |                | :star: |
+| compute pass           | :star: |        |        |           |        |        |                |        |
+| optional extension     |        |        |        |           |        |        | :star:         |        |
```

Comment:
do we actually want to list the optional things it supports? We could have something like
`| - binding indexing |` to show it's a sub item of the optional extensions.